### PR TITLE
Add `mint_eth` check on e2e desposit test

### PR DESCRIPTION
**Pull request by NiloCK**
PR "closes the loop" of the e2e L1-L2 deposits workflow by inspecting L2 for an emitted `rolluptypes.EventTypeMintETH` event.

## Summary by CodeRabbit

- **New Features**
	- Enhanced end-to-end testing capabilities with the addition of a function to verify ETH minting events.
  
- **Improvements**
	- Improved clarity in testing code through variable renaming for better context.

### CODE REVIEW

File: e2e/stack_test.go [hunk 1]
```diff
@@ -231,6 +234,37 @@ func TestE2E(t *testing.T) {
 	t.Log("Monomer blocks contain the l1 attributes deposit tx")
 }
 
+func requireEthIsMinted(t *testing.T, appchainClient *bftclient.HTTP) {
+	query := "tx.height > 0"
```
Comment (joshklop) [inline e2e/stack_test.go:238]: I think we can search for the specific event. The transaction indexer [indexes the events](https://docs.cometbft.com/v0.38/app-dev/indexing-transactions#adding-events) iirc.

File: e2e/stack_test.go [hunk 2]
```diff
@@ -155,20 +156,20 @@ func TestE2E(t *testing.T) {
 	)
 	require.NoError(t, err, "deposit tx")
 
-	client, err := bftclient.New(monomerCometURL.String(), monomerCometURL.String())
+	appchainClient, err := bftclient.New(monomerCometURL.String(), monomerCometURL.String())
```
Comment (joshklop) [inline e2e/stack_test.go:159]: purpose for the rename? I feel I'm missing the intent here.

**Review by joshklop — APPROVED** [review]
LGTM. Thanks!

File: e2e/stack_test.go (hunk 1 above)
Comment (NiloCK) [reply e2e/stack_test.go:238]: It does, but I burned nontrivial time trying to execute a more precise query.

For expediency, I'm inclined to leave this brutish check, which is still pretty readable.

File: e2e/stack_test.go (hunk 2 above)
Comment (NiloCK) [reply e2e/stack_test.go:159]: There are a few clients in this file to different processes (ethRPCClient, appchainClient, monomerClient). Disambiguating.

File: e2e/stack_test.go (hunk 1 above)
Comment (joshklop) [reply e2e/stack_test.go:238]: SGTM. Mind creating a "good first issue" for this? Would be a good educational exercise for new joiners.

Comment (joshklop) [reply e2e/stack_test.go:238]: Feel free to merge this PR whenever.

**Pull request merged**